### PR TITLE
feat(sdk-coin-atom): add sendMany support

### PR DESCRIPTION
**Pull request by dpkjnr**
Cosmos side chains natively support transactions with multiple recipients. This PR adds test cases to verify multiple recipient transaction building and fixes explain transaction code to generate input and outputs for all recipients.
Added tests only for Atom and Tia as Atom doesn't extend common cosmos classes, and adding tests for any one of other coins that extend cosmos base classes would suffice
Ticket: WIN-17

### CODE REVIEW

File: modules/sdk-coin-zeta/test/unit/zeta.ts [hunk 1]
```diff
@@ -268,7 +268,7 @@ describe('Zeta', function () {
             amount: 'UNAVAILABLE',
           },
         ],
-        outputAmount: 'UNAVAILABLE',
```
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-zeta/test/unit/zeta.ts:271]: any specific reason why we changed this ?

File: modules/sdk-coin-tia/test/unit/transactionBuilder/transferBuilder.ts [hunk 2]
```diff
@@ -157,4 +157,50 @@ describe('Tia Transfer Builder', () => {
...
+    txBuilder.chainId(testSendManyTx.chainId);
+    txBuilder.accountNumber(testSendManyTx.accountNumber);
+    txBuilder.addSignature(
+      { pub: toHex(fromBase64(testSendManyTx.pubKey)) },
+      Buffer.from(testSendManyTx.signature, 'base64')
+    );
+
+    const tx = await txBuilder.build();
+    const json = await (await txBuilder.build()).toJson();
+    should.equal(tx.type, TransactionType.Send);
```
Comment (Vijay-Jagannathan) [inline modules/sdk-coin-tia/test/unit/transactionBuilder/transferBuilder.ts:177]: Ideally we'd need this test for the other supported transaction types too

File: modules/abstract-cosmos/src/lib/transaction.ts [hunk 3]
```diff
@@ -174,64 +172,65 @@ export class CosmosTransaction extends BaseTransaction {
...
-          {
-            address: message.validatorAddress,
+        outputs = json.sendMessages.map((message) => {
+          const withdrawMessage = message.value as WithdrawDelegatorRewardsMessage;
+          return {
+            address: withdrawMessage.validatorAddress,
             amount: UNAVAILABLE_TEXT,
-          },
-        ];
-        outputAmount = UNAVAILABLE_TEXT;
```
Comment (Vijay-Jagannathan) [inline modules/abstract-cosmos/src/lib/transaction.ts:222]: was this intentionally removed ?

File: modules/sdk-coin-atom/test/resources/atom.ts [hunk 4]
```diff
@@ -44,6 +44,52 @@ export const TEST_SEND_TX = {
...
+        fromAddress: 'cosmos1smefkq9yaxgw9c9fhym6qsp746q884czeacn32',
+      },
+    },
+    {
+      typeUrl: '/cosmos.bank.v1beta1.MsgSend',
+      value: {
+        amount: [
+          {
+            denom: 'uatom',
+            amount: '500',
```
Comment (DinshawKothari) [inline modules/sdk-coin-atom/test/resources/atom.ts:79]: nit: can we use a txn with different amounts for the two recipients.

File: modules/sdk-coin-atom/test/unit/atom.ts [hunk 5]
```diff
@@ -209,6 +232,31 @@ describe('ATOM', function () {
       });
     });
 
+    it('should explain sendMany transfer transaction', async function () {
```
Comment (DinshawKothari) [inline modules/sdk-coin-atom/test/unit/atom.ts:235]: can we add one testcase for sendMany delegate and undelegate transfer type also.

File: modules/sdk-coin-tia/test/resources/tia.ts [hunk 6]
```diff
@@ -44,6 +44,51 @@ export const TEST_SEND_TX = {
...
+  accountNumber: 47998,
+  sequence: 45,
+  sendMessages: [
+    {
+      typeUrl: '/cosmos.bank.v1beta1.MsgSend',
+      value: {
+        amount: [
+          {
+            denom: 'utia',
+            amount: '100000',
```
Comment (DinshawKothari) [inline modules/sdk-coin-tia/test/resources/tia.ts:65]: same here, can we use different amounts.

File: modules/sdk-coin-atom/test/unit/atom.ts (hunk 5 above)
Comment (DinshawKothari) [reply modules/sdk-coin-atom/test/unit/atom.ts:235]: same for any abstract-cosmos coin also.

File: modules/sdk-coin-atom/src/lib/transaction.ts [hunk 7]
```diff
@@ -184,51 +183,53 @@ export class Transaction extends BaseTransaction {
    */
   explainTransactionInternal(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
     let outputs: TransactionRecipient[];
-    let message;
-    let outputAmount;
+    let outputAmount: bigint | undefined;
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-atom/src/lib/transaction.ts:186]: same comment here

File: modules/sdk-coin-zeta/test/unit/zeta.ts (hunk 1 above)
Comment (dpkjnr) [reply modules/sdk-coin-zeta/test/unit/zeta.ts:271]: I didn't see any usage of it and made is same as atom

File: modules/abstract-cosmos/src/lib/transaction.ts (hunk 3 above)
Comment (dpkjnr) [reply modules/abstract-cosmos/src/lib/transaction.ts:222]: yes, to make is similar to atom. I also checked that it is not being used anywhere

File: modules/sdk-coin-atom/test/resources/atom.ts (hunk 4 above)
Comment (dpkjnr) [reply modules/sdk-coin-atom/test/resources/atom.ts:79]: doing it for delegate and undelegate txns

File: modules/sdk-coin-tia/test/unit/transactionBuilder/transferBuilder.ts (hunk 2 above)
Comment (dpkjnr) [reply modules/sdk-coin-tia/test/unit/transactionBuilder/transferBuilder.ts:177]: Added to celestia

File: modules/abstract-cosmos/src/lib/transaction.ts [hunk 8]
```diff
@@ -174,64 +172,65 @@ export class CosmosTransaction extends BaseTransaction {
...
+          };
+        });
```
[the remark this answers is not in the document]
Comment (dpkjnr) [reply modules/abstract-cosmos/src/lib/transaction.ts:221]: yeah, for withdrawal txn amount is not passed

**Review by mullapudipruthvik — APPROVED** [review]
LGTM @DinshawKothari or @Vijay-Jagannathan can be final approver

**Review by DinshawKothari — APPROVED** [review]

**Pull request merged**